### DISCUSSION TITLE
各位，有谁知道enhanced-mode的取值，normal、fake-ip、redir-host三者的定义是什么？都是什么区别？

### DISCUSSION THREAD

**Discussion by TurnOffNOD** [MetaCubeX/mihomo#1361] [General]
各位，有谁知道enhanced-mode的取值，normal、fake-ip、redir-host三者的定义是什么？都是什么区别？文档里面没说啊。

clash有过说是移除redir-host的代码，但是这里取值又保留了，这是啥情况？clash meta又给加回来了？还是为了兼容性用了一个花活？

这文档说的不清楚啊。

**Comment by xiandu999-yu** [comment]
https://youtu.be/aKlH6KRt9Jc?t=806&si=hkI7xhnXjylKjAWX

**Comment by H2295** [comment]
我也想知道normal和redithost有什么区别

**Reply by xiandu999-yu** [reply]
normal我也想知道，至于redithost应该就是传统DNS模式吧？....

**Reply by TurnOffNOD** [reply]
> normal我也想知道，至于redithost应该就是传统DNS模式吧？....

其实redir-hos本身理解比较简单，就是传统dns模式；问题是clash那边说移除了这个模式，现在clash meta（mihomo）配置取值还保留着，文档又说的不清楚，是原始功能保留了，还是玩的花活？

**Comment by sleepm** [comment]
normal 是正常返回解析，使用配置的 dns 服务器
redir-host 是转发到节点去解析

**Reply by xiandu999-yu** [reply]
原来如此，所以normal是直接本地解析IP发VPS？

**Reply by TurnOffNOD** [reply]
> normal 是正常返回解析，使用配置的 dns 服务器 redir-host 是转发到节点去解析

其实redir-hos本身理解比较简单，就是传统dns模式；问题是clash那边说移除了这个模式，现在clash meta（mihomo）配置取值还保留着，文档又说的不清楚，是原始功能保留了，还是玩的花活？

**Comment by xiandu999-yu** [comment]
![image](https://github.com/MetaCubeX/mihomo/assets/58256481/f36616fa-6d49-45fe-b883-70a887c2748d)

**Reply by TurnOffNOD** [reply]
那么，现在版本更迭了这么多，名字改了，文档都大改过了，好多东西现在的文档里面已经没说了，这个解释还适用么？

**Reply by xiandu999-yu** [reply]
代码有改动吗？没有就还适用...

**Reply by TurnOffNOD** [reply]
> 代码有改动吗？没有就还适用...

为了用这个工具，先把代码阅读一遍是吧……

**Reply by xiandu999-yu** [reply]
> > 代码有改动吗？没有就还适用...
> 
> 为了用这个工具，先把代码阅读一遍是吧……

更新日志：那我走？
